akka-testkit/src/main/scala/akka/testkit/SocketUtil.scala

```diff
@@ -82,22 +82,27 @@ object SocketUtil {
 
         val address = hostname match {
           case RANDOM_LOOPBACK_ADDRESS =>
-            if (canBindOnAlternativeLoopbackAddresses) s"127.20.${Random.nextInt(256)}.${Random.nextInt(256)}"
+            // JDK limitation? You cannot bind on addresses matching the pattern 127.x.y.255,
+            // that's why the last component must be < 255
```

Comment:
Maybe include the findings that `-Djava.net.preferIPv4Stack=true` helps reproduce it and it might be related to https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/master/jdk/src/solaris/native/java/net/net_util_md.c#L1517-L1530 ?

Reply:
Though I guess picking x.y.z.255 never made sense since that is the broadcast address :/ https://tools.ietf.org/html/rfc922#section-7

Reply:
> Though I guess picking x.y.z.255 never made sense since that is the broadcast address :/ [tools.ietf.org/html/rfc922#section-7](https://tools.ietf.org/html/rfc922#section-7)

AFAIU, that depends on the subnet. Not sure, there is even a subnet for these localhost addresses (i.e. it's a /32 net consisting of a single host), so it seemed to work.